core/src/main/java/cucumber/runtime/model/CucumberFeature.java

```diff
@@ -71,7 +66,10 @@ private static void loadFromRerunFile(FeatureBuilder builder, ResourceLoader res
         for (Resource resource : resources) {
             String source = read(resource);
             if (!source.isEmpty()) {
-                featurePaths.addAll(Arrays.asList(source.split(" ")));
```

Comment:
With Gherkin v2 the filters were passed to the parser and the filtering was done there, so it was straightforward to add the line filters from the rerun file to the filters passed through the `load` method. Now with Gherkin v4 the filtering is instead done after the compiling of the features to pickles, so we are actually reading the rerun file twice, once to find the feature files to be parsed, and once to get the line filters to the pickle filter chain. This duplication was probably caused by that those two used was fixed at different times in the work or upgrading to Gherkin v4 (and the extract method refactoring missed).